api/v2/views/base.py

```diff
@@ -175,11 +175,17 @@ def perform_update(self, serializer):
                 instance = serializer.save(end_date=timezone.now(),
                                            membership=membership)
             else:
-                instance = serializer.save(end_date=timezone.now())
+                if self.request._method == "PATCH":
+                    instance = serializer.save(status=StatusType.objects.get(id=serializer.initial_data['status']))
+                else:
+                    instance = serializer.save(end_date=timezone.now())
```

Comment:
Actually, i see below you did put in end_date appropriately. Drop the 'end_date=' kwarg from this line.